src/System.Text.Json/tests/JsonStringTests.cs

```diff
@@ -110,6 +110,21 @@ public static void TestDateTimeOffset(string testStr, string expectedStr)
             Assert.Equal(dateTimeOffset, dateTimeOffset2);
         }
 
+        [Theory]
+        [MemberData(nameof(JsonDateTimeTestData.InvalidISO8601Tests), MemberType = typeof(JsonDateTimeTestData))]
+        public static void TestInvalidDateTime(string testStr)
+        {
+            var jsonString = new JsonString(testStr);
+
+            Assert.False(jsonString.TryGetDateTime(out DateTime dateTimeVal));
```

Comment:
Any way for there to be valid DateTime text that doesn't fit in 256 bytes when transcoded to UTF-8, so that `jsonString.TryGetDateTime` returns true? I don't think so, but something to verify (was looking for leading/trailing characters that might get ignored but they don't).

Even if the characters were escaped, I don't think it would return true.

Reply:
~In fact, I think we can be stricter than using `MaximumEscapedDateTimeOffsetParseLength` for the initial bounds check since in this context, escaped characters won't work anyway.~

~So, `IsValidDateTimeOffsetParseLength ` could use `MaxExpansionFactorWhileTranscoding * MaximumDateTimeOffsetParseLength`~

Hmm, so it looks like escaped characters don't work in the current implementation (and they should, similar to `JsonElement.GetDateTime` and `Utf8JsonReader.GetDateTime`. This is a bug here.

```C#
string testStr = "\"\\u0031997-07-16T19:20:30.4555555+14:00\"";

using var doc = JsonDocument.Parse(testStr);
Assert.True(doc.RootElement.TryGetDateTime(out DateTime dateTimeVal));

var jsonString = new JsonString(testStr);

Assert.True(jsonString.TryGetDateTime(out dateTimeVal));
```

Reply:
@khellang - given this is an existing bug, it's OK if you just want to focus on the `MaxExpansionFactorWhileTranscoding` fix in this PR. Let me know and I will file an issue instead.

Reply:
Filed an issue https://github.com/dotnet/corefx/issues/42582